lib/src/utils/format_tool_builder.dart

```diff
@@ -0,0 +1,233 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:dart_dev/src/tools/over_react_format_tool.dart';
+import 'package:glob/glob.dart';
+
+import '../../dart_dev.dart';
+import 'logging.dart';
+
+/// Visits a `dart_dev/config.dart` file and searches for a custom formatter.
+///
+/// When a custom formatter is found, this will look for possible configuration options
+/// and reconstruct them on a new formatter instance.
+///
+/// NOTE: Because the visitor doesn't have access to the scope of the configuration
+/// being parsed, most values need to be a literal type (ListLiteral, StringLiteral, IntegerLiteral)
+/// to be reconstructed.
+///
+/// Expects the configuration to be in the format of:
+/// ```dart
+/// final config = {
+///   'format': OverReactFormatTool()
+///     ..lineLength = 120,
+/// };
+/// ```
+class FormatToolBuilder extends GeneralizingAstVisitor<void> {
+  DevTool formatDevTool;
+
+  bool failedToDetectAKnownFormatter = false;
+
+  @override
+  visitMapLiteralEntry(MapLiteralEntry node) {
+    super.visitMapLiteralEntry(node);
+
+    String mapEntryKey = node.key.toSource().replaceAll("'", "");
+
+    if (mapEntryKey != 'format') return;
+
+    final formatterInvocation = node.value;
+    formatDevTool = detectFormatter(node.value);
+
+    if (formatDevTool == null) {
+      failedToDetectAKnownFormatter = true;
+      return;
+    }
+
+    if (formatterInvocation is CascadeExpression) {
+      List<Glob> reconstructedExcludesList = [];
+
+      final excludeExpression =
+          formatterInvocation.getCascadeByProperty('exclude');
+      if (excludeExpression != null) {
+        final detectedExcludesList = excludeExpression.rightHandSide;
+        if (detectedExcludesList is ListLiteral) {
+          reconstructedExcludesList = detectedExcludesList.elements
+              .whereType<MethodInvocation>()
+              .where((e) {
+            return e.argumentList.arguments.length == 1 &&
+                e.argumentList.arguments.first is StringLiteral;
+          }).map((e) {
+            final path =
+                (e.argumentList.arguments.first as StringLiteral).stringValue;
+            return Glob(path);
+          }).toList();
```

Comment:
I think this would miss excluded globs that use any of the [optional constructor params](https://pub.dev/documentation/glob/latest/glob/Glob/Glob.html). Maybe instead of filtering based on method invocations that only have 1 String arg, we could filter by the `Glob()` constructor name and only use the first arg?

Reply:
Just removed since we don't support it!